adminStuff/addNewPost.php

```diff
@@ -0,0 +1,35 @@
+<?php
+include('config/init.php');
+if(isset($_REQUEST['submitItem'])){
+    $errors = array();
+
+    if(!@$_REQUEST['blogTitle']){
+        $errors['blogTitle'] = "*Required*";
+    }
+
+    if(!@$_REQUEST['body']){
+        $errors['body'] = "*Required*";
+    }
+    if(!@$_REQUEST['tagName']){
+        $errors['tagName'] = "*Required*";
+    }
+    if(sizeof($errors) == 0){
+        echo addPostQuery($_REQUEST['blogTitle'], $_REQUEST['body'], $_REQUEST['tagName']);
+        exit;
+    }
+}
+
+echo echoAdminHeaderHtml("Admin Section", "Edit Post").
+        "</div><h2> New Post </h2><div class = 'forms'></br>
+            <form action='' method='post'>
+            <input type='hidden' name='dateOfPost' id='dateOfPost' value=".getPostDateTime(). " />".
+            singleLineTextBox('Blog Post Title', 'blogTitle','Blog Title', @$_REQUEST['blogTitle'], @$errors['blogTitle'])."
+            ".textareaTextBox ('Blog Post', 'body', 'Type blog post here...', @$_REQUEST['body'], @$errors['body'])."
```

Comment:
These functions look good. Deciding how much automatic stuff you want them to do is up to you (there's an argument for and against) but here's something to consider: could you make it so the functions don't need to be passed the request or the errors array and just automatically figure it out based on the name of the field you're passing in (the second parameter)?